src/twisted/words/protocols/irc.py

```diff
@@ -2620,9 +2622,17 @@ def connectionMade(self):
             self.register(self.nickname)
 
     def dataReceived(self, data):
-        basic.LineReceiver.dataReceived(self, data.replace('\r', ''))
+        data = data.replace('\r', '')
```

Comment:
I don't think this would work with a real TCP transport under Python 3, because the transport would call `dataReceived` with `bytes`, while `\r` and `''` are both native strings.